js_modules/dagit/packages/core/src/nav/useRepositoryLocationReload.tsx

```diff
@@ -79,37 +106,58 @@ export const useRepositoryLocationReload = (location: string) => {
         const workspace = data.workspaceOrError;
 
         if (workspace.__typename === 'PythonError') {
-          dispatch({type: 'error', error: workspace});
+          dispatch({type: 'error', error: workspace, errorLocationId: null});
           stopPolling();
           return;
         }
+        if (state.pollLocationIds === null) {
+          stopPolling();
+          return;
+        }
+
+        const matches = state.pollLocationIds
+          .map((location) => workspace.locationEntries.find((l) => l.id === location)!)
```

Comment:
Why does this need the `!`?

Reply:
I'm actually not sure, for some reason the `.filter(Boolean)` is not narrowing the TypeScript type to remove the undefined entries. Let me try a few things!